scripts/trace_flowcell.py

```diff
@@ -0,0 +1,118 @@
+        return start_time
+    #Compare elements in actions on key value start_timestamp.
+    sort_actions = sorted(actions, key=get_start_time)
+    return sort_actions
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Gets execution ids associated with a trace tag (e.g. a runfolder) and a "
+                                             "workflow. It can be used to track executions, e.g: "
+                                             "python scripts/trace_runfolder.py --tag 150605_M00485_0183_000000000-ABGT6_testbio14 | xargs -n1 st2 execution get")
+    parser.add_argument('--flowcell', required=True)
+    parser.add_argument('--api_base_url', default="https://localhost:4041/api/v1")
+    parser.add_argument('--workflow', default="workflow")
+    args = parser.parse_args()
+
+    api_base_url = args.api_base_url
+    workflow_name = args.workflow
+# Load config
+#with open("/opt/stackstorm/packs/snpseq_packs/config.yaml", 'r') as ymlfile:
+    with open("/home/riker752-local/projects/snpseq_packs/config.yaml", 'r') as ymlfile:
+        cfg = yaml.load(ymlfile)
+
+# Get hosts from config
+    hosts = cfg['runfolder_svc_urls']
+
+# Print header
+    print "\tstatus\trunfolder_link"
+
+    search_term = args.flowcell
+    hitcount = 0
+    choice = 0
+    folders = {}
+    for host in hosts:
+        url_base = '/'.join(host.split('/')[:-1])
+        result = requests.get("{}?state=*".format(url_base))
+        result_json = json.loads(result.text)
+        all_runfolders = result_json["runfolders"]
+        for runfolder in all_runfolders:
+            link = runfolder["link"]
+            state = runfolder["state"]
+            if search_term in link and not "rchive" in link and not "biotank" in link:
+                hitcount += 1
+                dirs = link.split('/')
+                folders[hitcount] = dirs[-1]
+                print "{}\t{}\t{}".format(hitcount, state, folders[hitcount])
+
+        if hitcount > 1:
+            choice = int(raw_input("Which runfolder to trace: "))
+        else:
+            choice = 1
+      
+        print "Will trace " + folders[choice]
+
+    # Try to load access token from environment - fall back if not available
+        try:
+            access_token = os.environ['ST2_AUTH_TOKEN']
+        except KeyError as e:
+            print("Could not find ST2_AUTH_TOKEN in environment. Please set it to your st2 authentication token.")
+
+        access_headers = {"X-Auth-Token": access_token}
+
+        traces = get_traces_for_tag(folders[choice], access_headers)
+        executions = get_executions_from_traces(traces, access_headers)
+    #print list(executions)
+        actions = get_actions_from_executions(executions, access_headers)
+        filtered_actions = filter_actions_by_name(actions, workflow_name)
+        sorted_actions = sort_actions_by_timestamp(filtered_actions)
+        for a in sorted_actions:
+            print a["id"]
```

Comment:
My suggestion is to change ``print a["id"]``
to
``            bashCommand = "st2 execution get ".format
            process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE) 
            output, error = process.communicate()
            print output.format(a["id"])``

to handle the piping concern in the next comment

Reply:
Personally, I would focus on working around the need for the interactive input. So perhaps instead of selecting which tag to trace, trace them all and indicate in the output which id belongs to which tag? Some input from the users, i.e. @MatildaAslin, is probably the best way to decide though as it entirely depends on how the output is supposed to be used..

Reply:
I think it's quite nice idea to call "st2 execution get" from within the script. We use the trace_runfolder script a lot and I would say that we in 99% of the cases pipe the output to `st2 execution get`.